File: sdk/php/src/Providers/TempFastMail.php
<?php

declare(strict_types=1);

namespace ChanhanzhanX\TempMail\Providers;

use ChanhanzhanX\TempMail\Email;
use ChanhanzhanX\TempMail\EmailInfo;
use ChanhanzhanX\TempMail\HttpClient;
use ChanhanzhanX\TempMail\Normalize;

/**
 * tempfastmail.com 渠道实现
 *
 * API: https://tempfastmail.com
 * 流程: POST /api/email-box 创建邮箱（返回 email + uuid，uuid 作为 token）→
 *       GET /api/email-box/{uuid}/emails 列表 → GET /api/email-box/{uuid}/email/{id} 详情。
 */
final class TempFastMail
{
    private const CHANNEL = 'tempfastmail';
    private const BASE_URL = 'https://tempfastmail.com';

    /** @var array<string,string> */
    private const HEADERS = [
        'Accept' => 'application/json',
        'User-Agent' => 'Mozilla/5.0',
    ];

    public static function generate(): EmailInfo
    {
        $resp = HttpClient::post(self::BASE_URL . '/api/email-box', self::HEADERS);
        $data = HttpClient::json($resp);
        $email = trim((string) ($data['email'] ?? ''));
        $uuid = trim((string) ($data['uuid'] ?? ''));
        if ($email === '' || !str_contains($email, '@') || $uuid === '') {
            throw new \RuntimeException('tempfastmail: 创建邮箱响应无效');
        }
        return new EmailInfo(self::CHANNEL, $email, $uuid);
    }

    /**
     * @return Email[]
     */
    public static function getEmails(string $email, ?string $token): array
    {
        $uuid = trim((string) $token);
        $address = trim($email);
        if ($uuid === '') {
            throw new \InvalidArgumentException('tempfastmail: token 不能为空');
        }
        if ($address === '') {
            throw new \InvalidArgumentException('tempfastmail: 邮箱地址不能为空');
        }

        $resp = HttpClient::get(self::BASE_URL . '/api/email-box/' . rawurlencode($uuid) . '/emails', self::HEADERS);
        $rows = HttpClient::json($resp);
        if (!array_is_list($rows)) {
            return [];
        }

        $out = [];
        foreach ($rows as $row) {
            $raw = is_array($row) ? $row : [];
            $messageId = trim((string) ($raw['uuid'] ?? ''));
            if ($messageId !== '') {
                $detailResp = HttpClient::get(
                    self::BASE_URL . '/api/email-box/' . rawurlencode($uuid) . '/email/' . rawurlencode($messageId),
                    self::HEADERS,
                );
                if ($detailResp->getStatusCode() < 400) {
                    $detail = HttpClient::json($detailResp);
                    if (!empty($detail)) {
                        $raw = $detail;
                    }
                }
            }
            $out[] = Normalize::email(self::flatten($raw, $address), $address);
        }
        return $out;
    }

    /**
     * @param array<mixed> $raw
     * @return array<string,mixed>
     */
    private static function flatten(array $raw, string $recipient): array
    {
        return [
            'id' => $raw['uuid'] ?? '',
            'from' => $raw['from'] ?? '',
            'to' => $raw['real_to'] ?? $recipient,
            'subject' => $raw['subject'] ?? '',
            'html' => $raw['html'] ?? '',
            'date' => $raw['received_at'] ?? '',
            'attachments' => $raw['attachments'] ?? [],
        ];
    }
}

File: sdk/csharp/src/Providers/TempFastMail.cs
using System.Collections.Generic;
using System.Net;
using System.Text.Json.Nodes;

namespace XxxXTeam.TempMail.Providers;

/// <summary>
/// TempFastMail 渠道 — https://tempfastmail.com。
/// 创建：POST /api/email-box 返回 email 与 uuid（uuid 作为 token）。
/// 收信：GET /api/email-box/{uuid}/emails 列表 + 逐封 GET /api/email-box/{uuid}/email/{id} 取详情。
/// </summary>
public static class TempFastMail
{
    private const string BaseUrl = "https://tempfastmail.com";
    private static readonly Dictionary<string, string> Headers = new()
    {
        ["Accept"] = "application/json",
        ["User-Agent"] = "Mozilla/5.0",
    };

    private static Dictionary<string, object?> Flatten(JsonObject raw, string recipient) => new()
    {
        ["id"] = Json.Str(raw, "uuid"),
        ["from"] = Json.Str(raw, "from"),
        ["to"] = !string.IsNullOrEmpty(Json.Str(raw, "real_to")) ? Json.Str(raw, "real_to") : recipient,
        ["subject"] = Json.Str(raw, "subject"),
        ["html"] = Json.Str(raw, "html"),
        ["date"] = Json.Str(raw, "received_at"),
        ["attachments"] = raw["attachments"] is JsonArray a ? Json.ToRaw(a) : null,
    };

    /// <summary>创建临时邮箱</summary>
    public static EmailInfo Generate()
    {
        var resp = Http.Post($"{BaseUrl}/api/email-box", null, null, new Dictionary<string, string>(Headers));
        resp.EnsureSuccess();
        var data = Json.Parse(resp.Body);
        var email = Json.Str(data, "email").Trim();
        var uuid = Json.Str(data, "uuid").Trim();
        if (string.IsNullOrEmpty(email) || !email.Contains('@') || string.IsNullOrEmpty(uuid))
            throw new System.Exception("tempfastmail: invalid create response");
        return new EmailInfo("tempfastmail", email, uuid);
    }

    /// <summary>获取邮件列表</summary>
    public static List<Email> GetEmails(string? token, string email)
    {
        var uuid = (token ?? "").Trim();
        var address = (email ?? "").Trim();
        if (string.IsNullOrEmpty(uuid)) throw new System.Exception("tempfastmail: empty token");
        if (string.IsNullOrEmpty(address)) throw new System.Exception("tempfastmail: empty email");

        var resp = Http.Get($"{BaseUrl}/api/email-box/{WebUtility.UrlEncode(uuid)}/emails", new Dictionary<string, string>(Headers));
        resp.EnsureSuccess();
        var rows = Json.Parse(resp.Body) as JsonArray;
        var result = new List<Email>();
        if (rows is null) return result;

        foreach (var r in rows)
        {
            var raw = r as JsonObject ?? new JsonObject();
            var messageId = Json.Str(raw, "uuid").Trim();
            if (!string.IsNullOrEmpty(messageId))
            {
                var dr = Http.Get($"{BaseUrl}/api/email-box/{WebUtility.UrlEncode(uuid)}/email/{WebUtility.UrlEncode(messageId)}",
                    new Dictionary<string, string>(Headers));
                if (dr.Ok && Json.Parse(dr.Body) is JsonObject detail) raw = detail;
            }
            result.Add(Normalize.NormalizeEmail(Flatten(raw, address), address));
        }
        return result;
    }
}
